### DISCUSSION TITLE
파이어베이스 페이징 처리 문제

### DISCUSSION THREAD

**Discussion by svclaw2000** [boostcampwm-2021/Android08-Ratatouille#96] [Issue Talk] [answered]
## 문제

Firestore에서 데이터 페이징하는 방법이 `컬럼A로 정렬하고 A값이 X 초과/미만인 값을 N개 가져온다`로 되어있습니다.

예를 들어, ViewCount가 1,2,3,4,5,6,7 이고 pageSize가 2, 현재 로드된 마지막 데이터의 ViewCount가 4면, ViewCount가 4보다 큰 데이터 2개를 가져온다. 이렇게 됩니다.

문제는, ViewCount가 10인 데이터가 10개 있고, 현재 로드된 마지막 데이터가 2번째 데이터면, 다음 페이지 요청할 때 응답으로 돌아오는 값이 3번 째 10의 데이터가 아니라, 그 다음 숫자(8 또는 9)의 데이터가 됩니다.

어떻게 해야 우리가 원하는대로 페이징을 처리할 수 있을까요....

**Comment by Jeong-heon2** [comment, marked as answer]
https://github.com/boostcampwm-2021/Android08-Ratatouille/pull/124
쿼리 스냅샷 으로 처리
